Example/Swift/ViewController.swift

```diff
@@ -188,31 +170,20 @@ class ViewController: UIViewController, MGLMapViewDelegate, AVSpeechSynthesizerD
         }
     }
     
-    func speak(_ text: String) {
-        let utterance = AVSpeechUtterance(string: text)
-        speechSynth.delegate = self
-        speechSynth.speak(utterance)
-    }
-    
     func startNavigation(_ route: Route) {
-        let camera = mapView.camera
-        camera.pitch = 40
-        mapView.setCamera(camera, animated: false)
-        mapView.userTrackingMode = .followWithCourse
-        navigation = RouteController(route: route)
-        navigation?.resume()
-    }
-    
-    func endNavigation() {
-        instructionView.isHidden = true
-        toggleNavigationButton.isHidden = true
-        howToBeginLabel.isHidden = false
-        mapView.userTrackingMode = .none
-        let camera = mapView.camera
-        camera.pitch = 0
-        mapView.setCamera(camera, animated: true)
-        removeRoutesFromMap()
-        navigation?.suspend()
+        // Pass through a
+        // 1. the route the user will take
+        // 2. A `Directions` class, used for rerouting.
+        let controller = NavigationUI.instantiate(route: route, directions: directions)
+        present(controller, animated: true, completion: nil)
+        
+        controller.willDismissNavigationHandler = {
```

Comment:
Forcing RouteViewController to own callbacks makes it all too easy for a developer to write a reference cycle that leaks memory. The two most common patterns for “handlers” are:

* Subclassing. If the application wants to customize the dismissal experience, it can subclass RouteViewController to override `dismiss(animated:completion:)` and `viewDidDisappear(_:)`.
* Delegation. Add a RouteViewControllerDelegate protocol that ViewController can conform to by implementing `routeViewControllerWillDismiss(_:)` and `routeViewControllerDidDismiss(_:)`.

Also, is there anything we can do to facilitate the use of an unwind segue to dismiss RouteViewController all the way to a starting point?